spec/index.js

```diff
@@ -122,3 +122,36 @@ describe('render()', function() {
         });
     });
 });
+
+describe('renderTheme()', function() {
+    const assembler = {
+        getTemplates: (path, processor) => {
+            return Promise.resolve(processor({
+                'pages/product': '<html>{{> pages/partial}}</html>',
+                'pages/partial': '<p>{{variable}}</p>',
+                'pages/greet': '<h1>{{lang \'good\'}} {{lang \'morning\'}}</h1>',
+                'pages/pre': '{{{pre object}}}',
+            }));
+        },
+        getTranslations: () => {
+            return Promise.resolve({});
+        }
+    };
+
+    const themeComponents = ['pages/product', 'pages/partial', 'pages/greet', 'pages/pre'];
+
+    it('should render theme', function(done) {
+        const paper = new Paper(null, null, assembler);
```

Comment:
Does this test cover the case we were seeing when the error was occurring?

Reply:
Yes, pages/partial is in pages/product and the p tags from it are part of the value for pages/product